easybuild/tools/filetools.py

```diff
@@ -2385,6 +2386,9 @@ def copy_file(path, target_path, force_in_dry_run=False):
             target_exists = os.path.exists(target_path)
             if target_exists and path_exists and os.path.samefile(path, target_path):
                 _log.debug("Not copying %s to %s since files are identical", path, target_path)
+            # Don't copy them if they are already considered the same
+            elif target_exists and path_exists and os.path.isfile(target_path) and filecmp.cmp(path, target_path):
```

Comment:
At the very least we should also check whether file permissions are the same as well before skipping the actual copy operation?

Reply:
Note that the permissions may have been changed later on by postinstallcmds or hooks or the normal EB change-permission step and thus permission differences should not be considered if the content is actually the same.